Maple2.Server.Game/PacketHandlers/ItemUseHandler.cs

```
@@ -106,4 +117,91 @@ public class ItemUseHandler : PacketHandler<GameSession> {
+        if (buddyBadgeBoxParams.Length != 2) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_unknown)));
+            Logger.Error("Invalid buddy badge box parameters: {Parameters}", item.Metadata.Function?.Parameters);
+            return;
+        }
+
+        if (!ItemMetadata.TryGet(buddyBadgeBoxParams[0], out ItemMetadata? itemMetadata)) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_unknown)));
+            Logger.Error("Unknown buddy badge box item Id: {Parameters}", buddyBadgeBoxParams[0]);
+            return;
+        }
+
+        var selfBadge = new Item(itemMetadata, buddyBadgeBoxParams[1]) {
+            CoupleInfo = new ItemCoupleInfo(receiverCharacterInfo.CharacterId, receiverCharacterInfo.Name, true),
+        };
+
+        if (!session.Item.Inventory.CanAdd(selfBadge)) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_gem_error_inventory_full)));
+            return;
+        }
+
+        if (!session.Item.Inventory.Consume(item.Uid, 1)) {
+            Logger.Error("Failed to use buddy badge box: {ItemUid}", item.Uid);
+            return;
+        }
+
+        var receiverMail = new Mail() {
+            ReceiverId = receiverCharacterInfo.CharacterId,
+            Type = MailType.System,
+            ContentArgs = new[] {
+                ("str", $"{session.Player.Value.Character.Name}"),
+            },
+        };
+
+        receiverMail.SetTitle(StringCode.s_couple_effect_mail_title_receiver);
+        receiverMail.SetContent(StringCode.s_couple_effect_mail_content_receiver);
+        receiverMail.SetSenderName(StringCode.s_couple_effect_mail_sender);
+
+        receiverMail = db.CreateMail(receiverMail);
+        if (receiverMail == null) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_unknown)));
```

Let's throw an exception here for now because this should never happen as the item was already consumed.